keyboards/skyloong/gk68/pro/keymaps/via/keymap.c

```diff
@@ -0,0 +1,172 @@
+        KC_CAPS,     KC_A,     KC_S,     KC_D,     KC_F,     KC_G,        KC_H,     KC_J,       KC_K,     KC_L,         KC_SCLN,     KC_QUOT,              KC_ENT,   KC_PGUP,
+        KC_LSFT,     KC_Z,     KC_X,     KC_C,     KC_V,     KC_B,        KC_N,     KC_M,    KC_COMM,   KC_DOT,         KC_SLSH,               KC_RSFT,     KC_UP,   KC_PGDN,
+        KC_LCTL,  KC_LALT,  KC_LGUI,             KC_SPC,                KC_SPC,               KC_SPC,  KC_RGUI,  MO(_MAC_INDEX),     KC_RALT,  KC_LEFT,   KC_DOWN,  KC_RIGHT
+    ),
+
+    [_WIN_INDEX] =  LAYOUT_all(
+         KC_GRV,    KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,       KC_F6,    KC_F7,      KC_F8,    KC_F9,          KC_F10,      KC_F11,   KC_F12,    KC_DEL,   KC_MUTE,
+        _______,  _______,    TO(0),    TO(1),  _______,  _______,     _______,  _______,    _______,  _______,         _______,    RGB_RMOD,  RGB_MOD,   RGB_TOG,   KC_MPLY,
+        _______,  _______,  _______,  _______,  _______,  _______,     _______,  _______,    _______,  _______,         _______,     _______,             _______,   KC_HOME,
+        _______,  _______,  _______,  _______,  _______,  _______,     _______,  _______,    _______,  _______,         _______,               _______,   RGB_VAI,    KC_END,
+        _______,  _______,  _______,            _______,               _______,              _______,  KC_RALT,         _______,      KC_APP,  RGB_SPD,   RGB_VAD,   RGB_SPI
+    ),
+
+    [_MAC_INDEX] = LAYOUT_all(
+        KC_GRV ,   KC_F14,   KC_F15, C(KC_UP),  G(KC_D), G(KC_SPC),  LSG(KC_4),  KC_MPRV,    KC_MPLY,  KC_MNXT,         KC_MUTE,     KC_VOLD,  KC_VOLU,    KC_DEL,   KC_MUTE,
+        _______,  _______,    TO(0),    TO(1),  _______,  _______,     _______,  _______,    _______,  _______,         _______,    RGB_RMOD,  RGB_MOD,   RGB_TOG,   KC_MPLY,
+        _______,  _______,  _______,  _______,  _______,  _______,     _______,  _______,    _______,  _______,         _______,     _______,             _______,   KC_HOME,
+        _______,  _______,  _______,  _______,  _______,  _______,     _______,  _______,    _______,  _______,         _______,               _______,   RGB_VAI,    KC_END,
+        _______,  KC_LALT,  KC_LGUI,            _______,               _______,              _______,  KC_RGUI,         _______,     KC_RALT,  RGB_SPD,   RGB_VAD,   RGB_SPI
+    )
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [_WIN] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(RGB_VAD, RGB_VAI)},
+    [_MAC] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(RGB_VAD, RGB_VAI)},
+    [_WIN_INDEX] = { ENCODER_CCW_CW(KC_MPRV, KC_MNXT), ENCODER_CCW_CW(RGB_HUD, RGB_HUI)},
+    [_MAC_INDEX] = { ENCODER_CCW_CW(KC_MPRV, KC_MNXT), ENCODER_CCW_CW(RGB_HUD, RGB_HUI)}
+};
+#endif
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
```

Comment:
This sort of functionality should really go in the keyboard level, as the default+via keymap should be pristine.

Reply:
OK, I do it.